agent/structs/config_entry_inline_certificate.go

```diff
@@ -84,6 +96,40 @@ func (e *InlineCertificateConfigEntry) Validate() error {
 	return nil
 }
 
+func validateKeyLength(privateKeyBlock *pem.Block) error {
+	if privateKeyBlock.Type != "RSA PRIVATE KEY" {
+		return nil
+	}
+
+	lenCheckFn := nonFipsLenCheck
+
+	if version.IsFIPS() {
+		lenCheckFn = fipsLenCheck
+	}
+	key, err := x509.ParsePKCS1PrivateKey(privateKeyBlock.Bytes)
+	if err != nil {
+		return err
+	}
+
+	return lenCheckFn(key.N.BitLen())
```

Comment:
Avoiding the function variable seems a bit more straightforward.

```suggestion
	key, err := x509.ParsePKCS1PrivateKey(privateKeyBlock.Bytes)
	if err != nil {
		return err
	}

	keyBits := key.N.BitLen()

	if version.IsFIPS() {
		return fipsLenCheck(keyBits)
	}
	return nonFipsLenCheck(keyBits)
```

Reply:
good call! I like this a lot more